core/jvm/src/main/scala/cats/effect/unsafe/WorkerThread.scala

```diff
@@ -317,6 +327,19 @@ private final class WorkerThread(
       }
     }
 
+    def parkUntilNextSleeper(): Unit = {
+      if (!isInterrupted()) {
+        val now = System.nanoTime()
+        val head = sleepersQueue.head()
+        val nanos = head.triggerTime - now
```

Comment:
cherry-pick https://github.com/typelevel/cats-effect/pull/3332/commits/e44a8023780da347a4e74a0b78ba581847e118eb
```suggestion
        val nanos = Math.max(head.triggerTime - now, 0)
```